### ISSUE TITLE
Element::attach_shadow's shadow host check is outdated

### ISSUE THREAD

**Issue by jdm** [servo/servo#34746]
Code: https://github.com/servo/servo/blob/2ab66ce67870acfff7bf18087421013475411380/components/script/dom/element.rs#L544
Spec: https://dom.spec.whatwg.org/#valid-shadow-host-name (step 4)
Tests: `./mach test-wpt tests/wpt/tests/shadow-dom/Element-interface-attachShadow.html`

**Comment by jdm** [comment]
Since we don't actually implement the declarative flag for shadow DOM yet, I don't think it makes sense to try to address this yet.

**Comment by longvatrong111** [comment]
Can we implement declarative shadow DOM now?
As my understanding, the steps are:

1. Update attributes of `template` element according to: [spec](https://html.spec.whatwg.org/multipage/scripting.html#template-contents).
2. Implement `slot` element.
3. Modify html5ever crate according to: [spec](https://html.spec.whatwg.org/#parsing-main-inhead,) make html5ever able to parse new attributes of `template`. 

Then we can implement declarative flag for ShadowRoot and update other methods which relate to  declarative shadow DOM.

**Comment by jdm** [comment]
Yes, we can start on that implementation whenever someone is available and interested in work on it!

**Comment by longvatrong111** [comment]
Please let me try this one.
